migrations/5_ico_crowdsale_deploy.js

```diff
@@ -0,0 +1,30 @@
+const MFC_Token = artifacts.require("MFC_Token");
+const WhitelistedCrowdsale = artifacts.require("WhitelistedCrowdsale");
+
+module.exports = function (deployer, network, accounts) {

```

Comment:
I think migration is too global. It is better to remove it and use the `beforeEach` method to instantiate the controller.
The method will look like this:

```
    let contract = null;

    beforeEach("setup contract for each test", async () => {
      let owner = accounts[0];
      let user = accounts[1];
      let token = await MFC_Token.new({from: owner});

      let startTime = Date.now();
      let endTime = startTime + 1000;
      let rate = 10;
      let minPurchase = 1;
      let maxPurchase = 1000;
      let wallet = user;
      let countPurchaseAmount = false;

      contract = await WhitelistedCrowdsale.new(
        startTime,
        endTime,
        rate,
        minPurchase,
        maxPurchase,
        wallet,
        token.address,
        countPurchaseAmount,
        {from: owner}
      );
    });
```

See the similar example here: https://github.com/MFChain/smart-contracts/blob/bf567aac209765c99f095e34a029f78fa9fe8be3/test/whitelisted_crowdsale.js#L29